code/core/src/manager/components/mobile/navigation/useFocusTrap.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useRef } from 'react';
+
+/**
+ * Hook to trap focus within a container when active
+ *
+ * @template T The type of the HTML element the ref will be attached to.
+ * @param isActive Whether the focus trap should be active
+ * @param onEscape Optional callback to run when Escape key is pressed
+ * @returns A ref to attach to the container element
+ */
+export function useFocusTrap<T extends HTMLElement = HTMLElement>(
+  isActive: boolean,
+  onEscape?: () => void
+): React.RefObject<T> {
+  const containerRef = useRef<T>(null);
+  const previousActiveElementRef = useRef<HTMLElement | null>(null);
+
+  // Store the active element when the trap becomes active and focus the container/first element
+  useEffect(() => {
+    const container = containerRef.current;
+    if (isActive && container) {
+      if (document.activeElement instanceof HTMLElement) {
+        previousActiveElementRef.current = document.activeElement;
+      }
+
+      const interactiveRoleElements = container.querySelectorAll<HTMLElement>(
+        'div[role="button"], span[role="button"], [role="link"]'
+      );
+      interactiveRoleElements.forEach((element) => {
+        if (!element.hasAttribute('tabindex')) {
+          element.setAttribute('tabindex', '0');
+        }
+      });
+
+      const focusableElements = getFocusableElements(container);
+
+      requestAnimationFrame(() => {
+        if (focusableElements.length > 0) {
+          focusableElements[0].focus();
+        } else {
+          // Make the container itself focusable if not already and no interactive elements
+          if (container.getAttribute('tabindex') === null || container.tabIndex < 0) {
+            container.setAttribute('tabindex', '-1'); // Or '0' if you want it in tab order when empty
+          }
+          container.focus();
+        }
+      });
+    } else if (!isActive && previousActiveElementRef.current) {
+      // Return focus to the element that was focused before the trap was activated
+      previousActiveElementRef.current.focus();
+      previousActiveElementRef.current = null; // Clear ref after returning focus
+    }
+  }, [isActive]);
+
+  // Set up the keyboard event listeners for the focus trap
+  useEffect(() => {
+    const container = containerRef.current;
+
+    if (!isActive || !container) {
+      return;
+    }
+
+    const handleKeyDown = (e: KeyboardEvent) => {
+      const focusableElements = getFocusableElements(container);
+
+      // If no focusable elements and not Escape, prevent default and exit
+      if (focusableElements.length === 0 && e.key !== 'Escape') {
+        e.preventDefault();
+        return;
+      }
```

Comment:
I wonder if that would break built-in keyboard shortcuts in screenreaders, e.g. those used to navigate between headings or paragraphs.